extensions/kubernetes/vanilla/deployment/src/main/java/io/quarkus/kubernetes/deployment/KubernetesDeployer.java

```diff
@@ -192,6 +196,21 @@ private DeploymentResultBuildItem deploy(DeploymentTargetEntry deploymentTarget,
 
     }
 
+    private void printExposeInformation(KubernetesClient client, KubernetesList list) {
+        List<HasMetadata> items = list.getItems();
+        for (HasMetadata item : items) {
+            if (Constants.ROUTE_API_GROUP.equals(item.getApiVersion()) && Constants.ROUTE.equals(item.getKind())) {
+                try {
+                    DefaultOpenShiftClient openShiftClient = new DefaultOpenShiftClient(client.getConfiguration());
+                    Route route = openShiftClient.routes().withName(item.getMetadata().getName()).get();
+                    log.infov("The deployed application can be accessed at: http://{0}", route.getSpec().getHost());
```

Comment:
Two things:

1. This would print all deployed routes. Most of the time, there will be just 1, but wouldn't it be better to print only the application's route? That is, the one with the same name as all the other generated resources.
2. If TLS is configured in the route, shouldn't we print `https://...`?

Reply:
Ah, also -- `new DefaultOpenShiftClient` for each deployed route?

And do we even need to create an `OpenShiftClient` when we already have the `KubernetesClient`?

Reply:
1. I don't see how that can happen - as the code reads the Route from the generated manifest
2. Good idea. I assume just checking if `tls` is part of the spec would suffice?

Reply:
> And do we even need to create an `OpenShiftClient` when we already have the `KubernetesClient`? Isn't it enough to `.adapt`?

This one is easy, I didn't know about it :P

Reply:
1. If I understand correctly, this can happen when you put "supplementary applications" into `src/main/kubernetes`.
2. I think so, yes.

Reply:
1. Interesting... I didn't think of that one. Let's be defensive

Reply:
Actually I think both approaches may make sense.

If there's just one route, print it the way you do now.
If there's more, than first print the "other" routes, with a slightly different message (e.g. "Deployment xxx can be accessed at: ..."), and print the application's route at the end, with current message.

Your call :-)

Reply:
Let's keep the logs are clutter free as possible :laughing: .

I went with only printing the application Route